cmd/ctr/events.go

```diff
@@ -22,39 +20,38 @@ var eventsCommand = cli.Command{
 		ctx, cancel := appContext(context)
 		defer cancel()
 
-		events, err := eventsClient.Subscribe(ctx, &eventsapi.SubscribeRequest{})
+		events, err := eventsClient.Subscribe(ctx, &eventsapi.SubscribeRequest{
+			Filters: context.Args(),
+		})
 		if err != nil {
 			return err
 		}
-		w := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
 		for {
 			e, err := events.Recv()
 			if err != nil {
 				return err
 			}
 
-			v, err := typeurl.UnmarshalAny(e.Event)
-			if err != nil {
-				return err
-			}
-			out, err := json.Marshal(v)
-			if err != nil {
-				return err
+			var out []byte
+			if e.Event != nil {
```

Comment:
You can't really trust that the remote side sends well-structured envelopes. More hardening needs to happen here, especially in the forwarder logic.